### DISCUSSION TITLE
[Bug report]  Pages are listed multiple times

### DISCUSSION THREAD

**Discussion by henkvantijen** [GetPublii/Publii#1760] [Bug Report]
### Operating system

macOS Sonoma 14.4.1

### Publii version

0.46.1

### Issue type

Application

### Bug description

After some editing, I see one page 3 times in the page listing and one page 7 times. Note the duplicate pages have the same Id. (17 and 19)
Selecting 1 page with checkbox, results in the checkboxes of each page getting selected.  And vv. 
So the entry in the list gets duplicated, but the page itself  is not duplicated, because one change in one of the duplicates appears in all listed similar pages. 
![puplii-dups-bug](https://github.com/user-attachments/assets/fd65ea26-b6b9-4771-afde-2355e0cc94d8)

Theme = Melissa 3.0.0.0

### Steps to reproduce

It is unclear what caused these duplicated to appear. So far, it only happened with pages that have an Image Gallery inserted. (But a test page with image gallery did not get duplicated). 

### Relevant log output

```shell
not found.
```

**Comment by michaelfluegge** [comment]
Please do not forget that we are using a version 0.46. ...
I would try making a back up of the site, de-install the app, restart the computer (depending on the system you use), install the app again and import the backup.

**Reply by henkvantijen** [reply]
Ok, I will try that. As I wrote, I am on MacOS, so 'uninstall'  is nothing more than dragging the app into the trash can, correct?

**Reply by michaelfluegge** [reply]
under MacOS Sonoma 14.7 you'll find parts of Publii here:
![Publii](https://github.com/user-attachments/assets/df49a24e-ecf1-409e-ac70-9a3fc0fc4aab)

**Comment by henkvantijen** [comment]
Danke! 🙏

**Comment by bjazmoore** [comment]
I had this issue when I upgraded a pre-0.46.x site to 0.46.0 and changes to a page enabled theme.  The problem only happened with pages.  I was at a loss as to why.  I uninstalled Publii and reinstalled but the problem did not go away.  Finally I completely deleted everything except my Publii backups and installed fresh (by this time 0.46.1 had been released).  That fixed my issue.  I could not reproduce the issue on my other PC's, so decided it was just a fluke.  It probably still is.

**Comment by dziudek** [comment]
@henkvantijen - did you upgraded your website as @bjazmoore described? In v.0.46.1 I have added some improvements to pages handling, but it seems that still something is missing. In theory - you can edit `input/config/pages.config.json` manually and delete duplicates. I will also check in the codebase how it is possible that duplicated IDs are added

**Comment by ProyectoFlower** [comment]
I have the same problem, but I think I made it worse because I tried to delete the duplicate and now it only appears as “????”. I uninstalled Publii completely (not without first backing it up) but it's still there, plus it causes the page menu to not work properly. I also checked the pages.config file and in theory everything looks fine, there are no duplicates.

**Reply by henkvantijen** [reply]
Question: the pages that appear duplicated, do the contain image galleries? I have an idea it may relate to galleries, because my pages without galeries are not duplicated. (Could be coincidence, o/c).

**Reply by ProyectoFlower** [reply]
Yes, so far the bug happens with the pages that have a gallery. It doesn't happen all the time though.

**Reply by henkvantijen** [reply]
@michaelfluegge   so maybe the page is saved (copied) in a loop for each image in a gallery ?  I'd search there.

**Comment by lgloaguen** [comment]
I have the same problem here. I'm a new user (0.46.2), so it's not a problem related to an update from a previous version.

I found myself with the same page duplicated in the list of pages. If I select one, the other is selected at the same time. They have the same ID.

Trying to solve the problem, I deleted the “double page” and created a new one with the same content. If this solved the double display on the page listing, I now have a “??” page, in the dropdown menu “Parent page”, which corresponds to the position of the deleted “double page”...

<img width="414" alt="publli-page" src="https://github.com/user-attachments/assets/9f60cea4-6d9f-4b16-9bde-7d4d6e749dbd">

I don't use the Image Gallery module at this time.

**Reply by ProyectoFlower** [reply]
The solution I could find is to go to the page.config.json file in the config folder in the input folder of your website and look for the duplicate page IDs and remove them.

**Reply by lgloaguen** [reply]
It worked for me. Thanks! I found and removed the reference to the deleted page in pages.config.json. I no longer have the “??” in the dropdown menu.

**Comment by carstenfeller** [comment]
Happened to me too. First version was 0.46.2, publii was upgraded to 0.46.3. Never run version 0.46.0 or 0.46.1.
Yesterday I created a page with only title, rest was blank, save and close.
Publii still running, I edited some posts.
Opened new page later again (still no duplicated entry in the pages list) and assigned a parent page.
Publii still running, I did some other stuff with posts and images.
Opened new page again (still no duplicate entry in the pages list) and set up content.
Made a backup and closed Publii, shut down PC.
Today I opened Publii and saw the page appearing two times in the pages list with same ID.
I removed the duplicate page ID from the pages.config.jason and everything is fine now.

**Comment by danielrpfeiffer** [comment]
Version: 0.46.4 (build 17047)

Experiencing the same issues.

Creating a site entirely in 0.46.4 but duplicate entries in the page listing and in the "Parent Page" Listing of individual pages start to clutter the listings.

removing duplicates from gages.config.json did remove most of the "??"-entries in the parent page listing but not all.

**Comment by dziudek** [comment]
@danielrpfeiffer - the issue will be fixed in v.0.46.5. The problem appears when you create a new page and click "Save" button in the editor multiple times. Fortunately somebody spot this behavior and I was able to fix it :)
